ebean-api/src/main/java/io/ebean/config/DatabaseConfig.java

```diff
@@ -905,6 +911,20 @@ public void setTenantCatalogProvider(TenantCatalogProvider tenantCatalogProvider
     this.tenantCatalogProvider = tenantCatalogProvider;
   }
 
+  /**
+   * Return true if dirty beans are automatically persisted.
+   */
+  public boolean isAutoPersistUpdates() {
+    return autoPersistUpdates;
```

Comment:
I think when transparent persistence is enabled (non-soft) deletes must be deferred and runs after updates (ideally in JDBC batch).

Scenario:
1. Update changes FK (deffered)
2. Delete removes FK referenced entity from DB (not deffered)
3.  FK constraint fails on DB

Flush order:
1. Insterts (or run immediatly)
2. Updates
3. Deletes

Reply:
Note that at this point this feature turns on JDBC batch - it is expected to run with JDBC batch turned on.

> deletes must be deferred

Probably not no. There are certainly cases where deletes must run before updates wrt OneToOne unique constraints etc. All the relative ordering is already handled in the existing cascade persist behaviour.  I don't see a change yet.

Reply:
I will make the PR with the test for the deferred delete use case.

Reply:
ok cool.